### PR TITLE
acronym: update test suite

### PR DESCRIPTION
**Pull request by cmccandless**
Bring test suite up-to-date with [canonical-data.json v1.7.0](https://github.com/exercism/problem-specifications/blob/cacf1f13cc1fd3bb18dee65fee99fd078dd8a541/exercises/acronym/canonical-data.json)

### CODE REVIEW

**Comment by cmccandless** [comment]
@SleeplessByte ready for review

File: exercises/acronym/example.js [hunk 1]
```diff
@@ -1 +1 @@
-export const parse = phrase => phrase.match(/[A-Z]+[a-z]*|[a-z]+/g).reduce((acronym, word) => `${acronym}${word[0]}`, '').toUpperCase();
+export const parse = phrase => phrase.replace("'", '').match(/^[A-Z]|(?<=[^A-Z])[A-Z]|\b[a-z]/g).join('').toUpperCase();
```
Comment (cmccandless) [inline exercises/acronym/example.js]: Might be worth breaking this into separate lines for readability.

Comment (SleeplessByte) [reply exercises/acronym/example.js]: Oh feel free! 

```javascript
export const parse = phrase => phrase
  .replace("'", '')
  .match(/^[A-Z]|(?<=[^A-Z])[A-Z]|\b[a-z]/g)
  .join('')
  .toUpperCase();
```

Comment (cmccandless) [reply exercises/acronym/example.js]:
```suggestion
export const parse = phrase => phrase
  .replace("'", '')
  .match(/^[A-Z]|(?<=[^A-Z])[A-Z]|\b[a-z]/g)
  .join('')
  .toUpperCase();
```

**Comment by SleeplessByte** [comment]
You're the best 🎉 🥇

**Pull request merged**